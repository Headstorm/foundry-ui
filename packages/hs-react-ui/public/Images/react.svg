<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
