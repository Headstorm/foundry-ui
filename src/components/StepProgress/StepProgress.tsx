import React from 'react';
import styled from 'styled-components';
import { darken } from 'polished';
import variants from '../../enums/variants';
import stepTypes from '../../enums/stepTypes';
import { getFontColorFromVariant } from '../../utils/color';
import { Div } from '../../htmlElements';
import fonts from '../../enums/fonts';
import { StyledSubcomponentType, SubcomponentPropsType } from '../commonTypes';
import { useTheme, useAnalytics } from '../../context';
import Button from '../Button';
import Text from '../Text';

export type ContainerProps = {
  disabled: boolean;
  vertical: boolean;
};
export const Container = styled(Div)`
  ${({ disabled, vertical }: ContainerProps) => `
    width: fit-content;
    position: relative;

    ${fonts.body}

    user-select: none;

    transition: filter .1s;

    ${
      disabled
        ? `
          filter: grayscale(1) contrast(.5) brightness(1.2);
          pointer-events: none;
        `
        : ''
    }
    ${
      vertical
        ? `
    transform-origin: top left;
    transform: translateX(50%) rotate(90deg);
    `
        : ''
    }
  `}
`;

const StepList = styled(Div)`
  position: relative;
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: stretch;
  flex-direction: row;
  gap: 1rem;
`;

const StepFlex = styled(Div)`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: space-between;
  flex: 1 1 0;
  min-width: 0;
`;

export const StepContainer = styled(Button.Container)`
  ${({ round, clickable }: { round: boolean; clickable: boolean }) => `
  ${
    round
      ? `border-radius: 100%;
      `
      : ''
  }
    ${
      clickable
        ? ''
        : `pointer-events: none;
      `
    }
    transition: background-color .3s, color .3s;
  `}
  width: 100%;
  &:before {
    display: block;
    content: '';
    height: 0;
    padding-bottom: 110%;
  }
`;

export const SelectedStepContainer = styled(StepContainer)`
  ${({ bgColor, borderColor }: { bgColor: string; borderColor: string }) => `
    color: ${getFontColorFromVariant(variants.fill, bgColor)};
    background-color: ${bgColor};
    border: 1px solid ${borderColor};
    &:hover {
      background-color: ${
        bgColor !== 'transparent' ? darken(0.05, bgColor) : 'rgba(0, 0, 0, 0.05)'
      };
    };
  `}
`;

export const NextStepContainer = styled(StepContainer)`
  ${({ bgColor, borderColor }: { bgColor: string; borderColor: string }) => `
    color: ${getFontColorFromVariant(variants.fill, bgColor)};
    background-color: ${bgColor};
    border: 1px solid ${borderColor};
    &:hover {
      background-color: ${
        bgColor !== 'transparent' ? darken(0.05, bgColor) : 'rgba(0, 0, 0, 0.05)'
      };
    };
  `}
`;

export const OutOfRangeStepContainer = styled(StepContainer)`
  ${({ bgColor, borderColor }: { bgColor: string; borderColor: string }) => `
    color: ${borderColor};
    background-color: ${bgColor};
    border: 1px solid ${borderColor};
    &:hover {
      background-color: ${
        bgColor !== 'transparent' ? darken(0.05, bgColor) : 'rgba(0, 0, 0, 0.05)'
      };
    };
  `}
`;

export const StepTextContainer = styled(Text.Container)`
  ${({ visible, vertical }: { visible: boolean; vertical: boolean }) => `
    display: inline-block;
    text-align: center;
    width: 100%;
    overflow-wrap: break-word;
    ${visible ? '' : 'visibility: hidden;'}
    ${vertical ? 'transform: rotate(-90deg);' : ''}
`}
`;

export const OverTextContainer = styled(StepTextContainer)`
  ${({ vertical }: { vertical: boolean }) => `
    margin-bottom: ${vertical ? '1.5rem' : '.5rem'};
    text-align: center;
    height: 100%;
    padding-bottom: 0;
    vertical-align: bottom;
  `}
`;

export const UnderTextContainer = styled(StepTextContainer)`
  ${({ vertical }: { vertical: boolean }) => `
    margin-top: ${vertical ? '1.5rem' : '.5rem'};
    height: 100%;
  `}
`;

export const SlideRail = styled(Div)`
  ${({ color }: { color: string }) => `
      position: absolute;
      top: 50%;
      transform: translateY(-50%);

      width: 100%;
      height: 0.25rem;

      overflow: hidden;

      border-radius: 0.125rem;
      background-color: ${color};
    `}
`;

export const SelectedRangeRail = styled(SlideRail)`
  ${({ index, max, color }: { index: number; max: number; color: string }) => {
    return `
      left: 0;
      width: ${100 * (index / (max - 1)) < 100 ? 100 * (index / (max - 1)) : 100}%;

      transition: width .3s;

      background-color: ${color};
    `;
  }}
`;

export type StepProgressProps = {
  StyledContainer?: StyledSubcomponentType;
  StyledSlideRail?: StyledSubcomponentType;
  StyledSelectedRangeRail?: StyledSubcomponentType;
  StyledNextStepContainer?: StyledSubcomponentType;
  StyledSelectedStepContainer?: StyledSubcomponentType;
  StyledOutOfRangeStepContainer?: StyledSubcomponentType;
  StyledCompletedStepContainer?: StyledSubcomponentType;
  StyledInnerTextContainer?: StyledSubcomponentType;
  StyledOverTextContainer?: StyledSubcomponentType;
  StyledUnderTextContainer?: StyledSubcomponentType;

  containerProps?: SubcomponentPropsType;
  completedStepProps?: SubcomponentPropsType;
  selectedStepProps?: SubcomponentPropsType;
  nextStepProps?: SubcomponentPropsType;
  outOfRangeStepProps?: SubcomponentPropsType;
  allStepsProps?: SubcomponentPropsType;
  textProps?: SubcomponentPropsType;

  containerRef?: React.RefObject<HTMLDivElement>;
  buttonRefs?: React.RefObject<HTMLButtonElement>[];
  stepRefs?: React.RefObject<HTMLDivElement>[];

  index?: number;
  steps?: string[];
  onClicks?: { (): void }[];

  stepType?: string;
  round?: boolean;
  vertical?: boolean;
  completeColor?: string;
  incompleteColor?: string;
  selectedStepColor?: string;
  nextStepColor?: string;
  canClickToNextStep?: boolean;
  canClickToPreviousSteps?: boolean;
  showSlideRail?: boolean;
  clickable?: boolean;
  disabled?: boolean;
};

export const StepProgress = ({
  StyledContainer = Container,
  StyledSlideRail = SlideRail,
  StyledSelectedRangeRail = SelectedRangeRail,
  StyledSelectedStepContainer = SelectedStepContainer,
  StyledNextStepContainer = NextStepContainer,
  StyledOutOfRangeStepContainer = OutOfRangeStepContainer,
  StyledCompletedStepContainer = StepContainer,
  StyledInnerTextContainer = StepTextContainer,
  StyledOverTextContainer = OverTextContainer,
  StyledUnderTextContainer = UnderTextContainer,

  containerProps = {},
  completedStepProps = {},
  selectedStepProps = {},
  nextStepProps = {},
  outOfRangeStepProps = {},
  allStepsProps = {},
  textProps = {},

  containerRef,
  buttonRefs = [],
  stepRefs = [],

  index = 0,
  steps = [],
  onClicks = [],

  stepType = stepTypes.inner,
  round = false,
  vertical = false,
  completeColor,
  incompleteColor,
  selectedStepColor = '#fff',
  nextStepColor,
  canClickToNextStep = true,
  canClickToPreviousSteps = true,
  showSlideRail = true,
  clickable = true,
  disabled = false,
}: StepProgressProps): JSX.Element | null => {
  const { colors } = useTheme();
  const handleEventWithAnalytics = useAnalytics();

  const finishedColor = completeColor || colors.primaryDark;
  const unfinishedColor = incompleteColor || colors.grayXlight;
  const nextStepFinalColor = nextStepColor || colors.grayXlight;

  const defaultCompletedStepProps: SubcomponentPropsType = {
    bgColor: finishedColor,
    round,
    clickable: clickable && canClickToPreviousSteps,
    stepType,
  };

  const defaultSelectedStepProps: SubcomponentPropsType = {
    bgColor: selectedStepColor,
    borderColor: finishedColor,
    round,
    clickable: false,
    stepType,
  };

  const defaultNextStepProps: SubcomponentPropsType = {
    bgColor: nextStepFinalColor,
    borderColor: unfinishedColor,
    round,
    clickable: true,
    stepType,
  };

  const defaultOutOfRangeStepProps: SubcomponentPropsType = {
    bgColor: '#fff',
    borderColor: unfinishedColor,
    round,
    clickable: false,
    stepType,
  };

  const getTextColor = (i: number) => {
    if (index === i) {
      return finishedColor;
    }
    if (index === i - 1) {
      if (canClickToNextStep) {
        return colors.grayMedium;
      }
      return colors.grayXlight;
    }
    if (index < i) {
      return colors.grayXlight;
    }
    return '#000';
  };

  const getContainer = (i: number) => {
    if (index === i) {
      return StyledSelectedStepContainer;
    }
    // The next step if clickable
    if (i === index + 1 && clickable && canClickToNextStep) {
      return StyledNextStepContainer;
    }
    // Anything past the selected step
    if (i > index) {
      return StyledOutOfRangeStepContainer;
    }
    // Previous steps
    return StyledCompletedStepContainer;
  };

  const getContainerProps = (i: number) => {
    // Selected step
    if (index === i) {
      return { ...defaultSelectedStepProps, ...selectedStepProps, ...allStepsProps };
    }
    // The next step
    if (i === index + 1 && clickable && canClickToNextStep) {
      return {
        ...defaultNextStepProps,
        ...nextStepProps,
        ...allStepsProps,
      };
    }
    // Anything past the selected step
    if (i > index) {
      return { ...defaultOutOfRangeStepProps, ...outOfRangeStepProps, ...allStepsProps };
    }
    // Previous steps
    return { ...defaultCompletedStepProps, ...completedStepProps, ...allStepsProps };
  };

  return (
    <StyledContainer ref={containerRef} disabled={disabled} vertical={vertical} {...containerProps}>
      {showSlideRail ? (
        <>
          <StyledSlideRail color={unfinishedColor} />
          <StyledSelectedRangeRail index={index} max={steps.length} color={finishedColor} />
        </>
      ) : (
        ''
      )}
      <StepList>
        {steps.map((step, i) => (
          <StepFlex ref={stepRefs[i]} key={`${step}-step`}>
            <Text
              StyledContainer={StyledOverTextContainer}
              containerProps={{ visible: stepType === stepTypes.over, vertical, ...textProps }}
              color={getTextColor(i)}
            >
              {step}
            </Text>
            <Button
              containerRef={buttonRefs[i]}
              color={String(getContainerProps(i).bgColor)}
              StyledContainer={getContainer(i)}
              containerProps={getContainerProps(i)}
              variant={variants.fill}
              onClick={
                clickable &&
                ((index < i && canClickToNextStep) || (index >= i && canClickToPreviousSteps))
                  ? () =>
                      handleEventWithAnalytics(
                        'StepProgress',
                        onClicks[i],
                        'onClick',
                        { type: 'onClick', clickIndex: i },
                        containerProps,
                      )
                  : () => {}
              }
            >
              <Text
                StyledContainer={StyledInnerTextContainer}
                containerProps={{ visible: true, vertical, ...textProps }}
              >
                {stepType === stepTypes.inner ? step : i}
              </Text>
            </Button>
            <Text
              StyledContainer={StyledUnderTextContainer}
              containerProps={{ visible: stepType === stepTypes.under, vertical, ...textProps }}
              color={getTextColor(i)}
            >
              {step}
            </Text>
          </StepFlex>
        ))}
      </StepList>
    </StyledContainer>
  );
};

StepProgress.stepTypes = stepTypes;
StepProgress.Container = Container;
StepProgress.SlideRail = SlideRail;
StepProgress.SelectedRangeRail = SelectedRangeRail;
StepProgress.StyledSelectedStepContainer = SelectedStepContainer;
StepProgress.StyledNextStepContainer = NextStepContainer;
StepProgress.StyledOutOfRangeStepContainer = OutOfRangeStepContainer;
StepProgress.StyledCompletedStepContainer = StepContainer;
StepProgress.StyledInnerTextContainer = StepTextContainer;
StepProgress.StyledOverTextContainer = OverTextContainer;
StepProgress.StyledUnderTextContainer = UnderTextContainer;

export default StepProgress;
