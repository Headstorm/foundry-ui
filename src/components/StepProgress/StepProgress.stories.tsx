import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import { Story, Meta } from '@storybook/react';
import colors from '../../enums/colors';
import stepTypes from '../../enums/stepTypes';
import StepProgress, { StepProgressProps } from './StepProgress';
import fonts from '../../enums/fonts';

type DefaultProps = StepProgressProps;

const Row = styled.div`
  ${fonts.body}
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  padding: 0.5rem;
  min-width: 20rem;
  width: 90%;
  margin: 0 auto;
`;

export const Default: Story<DefaultProps> = ({ ...args }: DefaultProps) => {
  const [currStep, setCurrStep] = useState(args.index);

  useEffect(() => setCurrStep(args.index), [args.index]);

  const onClicks: any[] = [];
  args.steps = args.steps || [];
  args.steps.forEach((step, i) => {
    onClicks.push(() => setCurrStep(i));
  });

  return (
    <Row>
      <StepProgress
        onClicks={onClicks}
        disabled={args.disabled}
        showSlideRail={args.showSlideRail}
        steps={args.steps}
        index={currStep}
        completeColor={args.completeColor}
        incompleteColor={args.incompleteColor}
        selectedStepColor={args.selectedStepColor}
        nextStepColor={args.nextStepColor}
        clickable={args.clickable}
        round={args.round}
        vertical={args.vertical}
        canClickToNextStep={args.canClickToNextStep}
        canClickToPreviousSteps={args.canClickToPreviousSteps}
        stepType={args.stepType}
      />
    </Row>
  );
};
Default.args = {
  disabled: false,
  showSlideRail: true,
  index: 1,
  completeColor: colors.primaryDark,
  incompleteColor: colors.grayXlight,
  selectedStepColor: '#fff',
  nextStepColor: colors.grayXlight,
  clickable: true,
  round: false,
  vertical: false,
  steps: ['Step 1', 'Step 2', 'Step 3', 'Step 4'],
  canClickToNextStep: true,
  canClickToPreviousSteps: true,
  stepType: stepTypes.inner,
};

export default {
  title: 'StepProgress',
  argTypes: {
    index: {
      control: {
        type: 'range',
        min: 0,
        max: 10,
        step: 1,
      },
    },
    stepType: {
      options: [stepTypes.inner, stepTypes.under, stepTypes.over],
      control: {
        type: 'select',
      },
    },
  },
} as Meta;
