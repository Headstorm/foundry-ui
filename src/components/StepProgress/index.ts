import StepProgress from './StepProgress';

export default StepProgress;
