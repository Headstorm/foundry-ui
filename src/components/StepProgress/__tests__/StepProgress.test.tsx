import React from 'react';
import { render, fireEvent, waitFor, configure } from '@testing-library/react';
import StepProgress from '../StepProgress';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);
configure({ testIdAttribute: 'data-test-id' });

const testId = 'hs-ui-step-progress-unit-test';

describe('RangeSlider', () => {
  it('renders', async () => {
    const { container, getByTestId } = render(
      <StepProgress
        index={1}
        labels={['0', '1', '2']}
        containerProps={{ 'data-test-id': testId }}
      />,
    );

    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('renders', async () => {
    const { container, getByTestId } = render(
      <StepProgress
        index={1}
        labels={['0', '1', '2']}
        containerProps={{ 'data-test-id': testId }}
      />,
    );

    await waitFor(() => getByTestId(testId));
    fireEvent.mouseDown(getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  describe('Accessibility Tests', () => {
    it('Should pass accessibility test with default props', async () => {
      const component = (
        <StepProgress
          index={1}
          labels={['0', '1', '2']}
          containerProps={{ 'data-test-id': testId }}
        />
      );
      const { container } = render(component);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
  describe('Ref tests', () => {
    it('containerRef.current should exist', async () => {
      const ref = React.createRef<HTMLDivElement>();
      const { getByTestId } = render(
        <StepProgress
          containerRef={ref}
          index={1}
          labels={['0', '1', '2']}
          containerProps={{ 'data-test-id': testId }}
        />,
      );
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLDivElement).toBeTruthy();
    });
  });
});
