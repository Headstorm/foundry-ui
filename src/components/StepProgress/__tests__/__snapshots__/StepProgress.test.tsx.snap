// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider renders 1`] = `
.c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  position: relative;
  font-family: Arial,Roboto,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: filter .1s;
  transition: filter .1s;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
}

.c1 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
  height: 0.25rem;
  overflow: hidden;
  border-radius: 0.125rem;
  background-color: #D5D5D5;
}

.c2 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
  height: 0.25rem;
  overflow: hidden;
  border-radius: 0.125rem;
  background-color: #BF3A00;
}

.c3 {
  left: 0;
  width: -100%;
  -webkit-transition: width .3s;
  transition: width .3s;
  background-color: #BF3A00;
}

<div>
  <div
    class=" c0"
    data-test-id="hs-ui-step-progress-unit-test"
  >
    <div
      class=" c1"
      color="#D5D5D5"
    />
    <div
      class=" c2 c3"
      color="#BF3A00"
      max="0"
    />
    <div
      class=" c4"
    />
  </div>
</div>
`;

exports[`RangeSlider renders 2`] = `
.c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  position: relative;
  font-family: Arial,Roboto,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: filter .1s;
  transition: filter .1s;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
}

.c1 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
  height: 0.25rem;
  overflow: hidden;
  border-radius: 0.125rem;
  background-color: #D5D5D5;
}

.c2 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
  height: 0.25rem;
  overflow: hidden;
  border-radius: 0.125rem;
  background-color: #BF3A00;
}

.c3 {
  left: 0;
  width: -100%;
  -webkit-transition: width .3s;
  transition: width .3s;
  background-color: #BF3A00;
}

<div>
  <div
    class=" c0"
    data-test-id="hs-ui-step-progress-unit-test"
  >
    <div
      class=" c1"
      color="#D5D5D5"
    />
    <div
      class=" c2 c3"
      color="#BF3A00"
      max="0"
    />
    <div
      class=" c4"
    />
  </div>
</div>
`;
